tests/application_development/code_relocation/src/main.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ * Copyright (c) 2022, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <ztest.h>
+
+/* This function will allow execute from sram region.
+ * This is needed only for this sample because by default all soc will
+ * disable the execute from SRAM.
+ * An application that requires that the code be executed from SRAM will
+ *  have to configure the region appropriately in arm_mpu_regions.c.
+ */
+
+#if (defined(CONFIG_ARM_MPU) && !defined(CONFIG_CPU_HAS_NXP_MPU))
+#include <zephyr/arch/arm/aarch32/cortex_m/cmsis.h>
```

Comment:
@povergoing , I did not find any exported API in arm_mpu.h which allows me to chnage the MPU setting in the runtime, and only throught the cmsis.h that I can access the register

Reply:
`arm_mpu.h` includes `arm_mpu_v7m.h` or `arm_mpu_v8.h` depends on the config and `arm_mpu_v7m.h` includes the `cmsis.h`.
Okay, I checked this test is only for Cortex-M, I would not persist using `arm_mpu.h`. I mentioned the `arm_mpu.h` because I thought it was both for M and A/R.